UserListScreen.tsx

```diff
@@ -39,91 +39,112 @@ interface User {
-              </View>
-            }
-            subtitle={
-              <View style={styles.content}>
-                  <Text style={styles.text}>{item.email}</Text>
-              </View>
-            }
-          />
+      <ListItem
+        key={ item.id }
+        title={
+          <View style={styles.content}>
+              <Text style={styles.text}>{item.name}</Text>
+          </View>
+        }
+        subtitle={
+          <View style={styles.content}>
+              <Text style={styles.text}>{item.email}</Text>
+          </View>
+        }
+      />
     );
 
     getUsersQuery = gql`
-      query Users {
-        Users {
-          count,
+      query Users($limit: Int, $offset: Int) {
+        Users(limit: $limit, offset: $offset) {
           nodes {
             id,
             name,
             email
           }
+          pageInfo {
+            hasNextPage
+          }
         }
       }`;
 
     private fillUpList(data: Response, error: ApolloError, loading: boolean): JSX.Element {
-        return (
-            loading ?
-                <View style={{
-                  flex: 1,
-                  justifyContent: 'center',
-                  alignItems: 'center'
-                }} >
-                  <ActivityIndicator 
-                    size="large"
-                    color="#0000ff"
-                    style={{ 
-                        zIndex: 0,
-                        position: 'absolute'
-                    }} />
-                </View>
-
-            : error ?
-              <View style={{
-                flex: 1,
-                justifyContent: 'center',
-                alignItems: 'center'
-              }} >
-                <Text style={{ color: "#FF0000", fontSize: 25, fontWeight: 'bold' }}>
-                    Erro interno do servidor
-                </Text>
-              </View>
-            
-            :
-              <FlatList
-                keyExtractor={this.keyExtractor}
-                data={data.Users.nodes}
-                renderItem={this.renderItem}
-              />
-        );
+      this.fetchingFromServer = loading;
+      if (data && !loading) {
+        this.hasNextPage = data.Users.pageInfo.hasNextPage;
+        this.currentListData = this.currentListData.concat(data.Users.nodes);
+      }
+      return (
+        error ?
+          <View style={{
+            flex: 1,
+            justifyContent: 'center',
+            alignItems: 'center'
+          }} >
+            <Text style={{ color: "#FF0000", fontSize: 25, fontWeight: 'bold' }}>
+                Erro interno do servidor
+            </Text>
+          </View>
+        
+        :
+          <FlatList
+            keyExtractor={this.keyExtractor}
+            data={this.currentListData}
+            renderItem={this.renderItem}
+            ItemSeparatorComponent={() => <View style={styles.separator} />}
+            ListFooterComponent={this.renderFooter.bind(this)}
```

Comment:
Aqui você pode tomar um atalho e enviar só a referência da função: 

```jsx
<FlatList
  ListFooterComponent={this.renderFooter}
/>
```

Mas isso só funciona se o `renderFooter` for uma variável de instância em vez de uma função, senão o `this` perde a referência:

```tsx
private renderFooter = () => {
  // Código do footer
}
```

Reply:
Obrigado pelo detalhe da variável de instância ;)
Corrigido